runtime/tests/org.eclipse.core.tests.runtime/src/org/eclipse/core/tests/runtime/jobs/JobTest.java

```diff
@@ -1602,6 +1620,38 @@ protected IStatus run(IProgressMonitor monitor) {
 		group.done();
 	}
 
+	public void testSetProgressGroup_progressGroupIsCanceled_monitorInJobIsCanceled() throws InterruptedException {
+		Job checkMonitorIsCanceled = createJobThatChecksItsOwnMonitor();
+
+		final IProgressMonitor progressGroup = Job.getJobManager().createProgressGroup();
+		progressGroup.beginTask("", 2);
+
+		checkMonitorIsCanceled.setProgressGroup(progressGroup, 1);
+
+		// let the job do the cancellation
+		progressGroup.setCanceled(true);
+
+		checkMonitorIsCanceled.schedule();
+		checkMonitorIsCanceled.join();
+
+		assertEquals("The job should have been canceled", IStatus.CANCEL, checkMonitorIsCanceled.getResult().getSeverity());
+	}
+
+	private Job createJobThatChecksItsOwnMonitor() {
```

Comment:
nit: this naming sounds as if it was some special case that a job checks its own monitor but actually that's "usualy behavior". Isn't the purpose of the job simply to react to a cancel request, i.e., `createJobReactingToCancelRequest`?